prow/tide/tide.go

```diff
@@ -103,16 +103,23 @@ func toSimpleState(s kube.ProwJobState) simpleState {
 	return noneState
 }
 
-func pickSmallestNumber(prs []pullRequest) pullRequest {
+func pickSmallestPassingNumber(prs []pullRequest) (bool, pullRequest) {
 	smallestNumber := -1
 	var smallestPR pullRequest
 	for _, pr := range prs {
-		if smallestNumber < 0 || int(pr.Number) < smallestNumber {
-			smallestNumber = int(pr.Number)
-			smallestPR = pr
+		if smallestNumber != -1 && int(pr.Number) >= smallestNumber {
+			continue
+		}
+		if len(pr.Commits.Nodes) < 1 {
+			continue
 		}
+		if string(pr.Commits.Nodes[0].Commit.Status.State) != "SUCCESS" {
```

Comment:
`// TODO: match against list of required status checks`

Reply:
Done.